src/app/ReadClient.h

```diff
@@ -251,6 +262,8 @@ class ReadClient : public Messaging::ExchangeDelegate
     void CancelLivenessCheckTimer();
     void MoveToState(const ClientState aTargetState);
     CHIP_ERROR ProcessAttributePath(AttributePathIB::Parser & aAttributePath, ClusterInfo & aClusterInfo);
```

Comment:
Please add a helper method to permit safe decoding of an event given a templated type that points to the cluster object for that event, the event header + reader, and a reference to the templated object. The method will validate that the Event ID and Cluster ID in the header match that of the type information present in the object before decoding the data.

Reply:
added, thanks